Adds option to perform EFA ordinal variables and a couple other options (Mardia's tests; Parallel Analysis detailed output) (#75)
jasp-stats/jaspFactor: inst/help/ExploratoryFactorAnalysis.md

Exploratory Factor Analysis 
=== 

With Exploratory Factor Analysis it is possible to identify one or more factors underlying the data. The factors are chosen such that they capture the common variance in the data.

### Assumptions (Yong & Pearce, 2013)
- The variables included in the analysis are continuous*.
  - If variables are ordinal, this assumption can be overcome when basing the analysis on the polychoric or tetrachoric correlation matrix (Bandalos & Finney, 2018).
- The data follow a multivariate normal distribution.
- There is a linear relation between the variables and the factors.
- There is no multicollinearity and singularity in the data.

### Input
---
#### Asssignment Box 
- Included Variables: In this box, the variables to perform the exploratory factor analysis on are selected. 

#### Number of Factors 
- _NB: eigenvalues for EFA are different from eigenvalues for PCA. See Dinno (2014) for more information._
- Here, the number of factors that the rotation is applied to is specified. Several methods to determine this number can be chosen from:   

>   - Parallel Analysis: Factors are selected on the basis of parallell analysis. With this method, factors are selected when their eigenvalue is greater than the parallel average random eigenvalue. This method is selected by default. 

Reviewer: ```suggestion
  - Parallel Analysis: Factors are selected on the basis of parallel analysis. With this method, factors are selected when their eigenvalue is greater than the parallel average random eigenvalue. This method is selected by default. Can be based on principal component eigenvalues (PC) or factor eigenvalues (FA).
```

- Include Tables: 
    - Factor correlations: When selecting this option, a table with the correlations between the factors will be displayed. 
    - Additional fit indices: This option displays the Root Mean Squared Error of Approximation (RMSEA) with 90% confidence interval, the Tucker Lewis Index (TLI), and the Bayesian Information Criterion (BIC) to test the fit of the model. 

>     - Parallel analysis: If this option is selected, a table will be generated exhibiting a detailed output of the parallel analysis.

Reviewer: ```suggestion
    - Parallel analysis: If this option is selected, a table will be generated exhibiting a detailed output of the parallel analysis. Can be based on principal component eigenvalues (PC) or factor eigenvalues (FA).
```

    - Path diagram: By selecting this option, a visual representation of the direction and strength of the relation between the variable and factor will be displayed. 
    - Scree plot: When selecting this option, a scree plot will be displayed. The scree plot provides information on how much variance in the data, indicated by the eigenvalue, is explained by each factor. A scree plot can be used to decide how many factors should be selected.
- Assumption Checks:

> 	- Mardia's Test of Multivariate Normality: Assesses the degree of the departure from multivariate normality of the included variables in terms of multivariate skewness and kurtosis.

Reviewer: ```suggestion
	- Mardia's Test of Multivariate Normality: Assesses the degree of the departure from multivariate normality of the included variables in terms of multivariate skewness and kurtosis. The Maria's test will always include the listwise complete cases.
```

  - RMSEA 90% confidence interval: The 90% confidence interval of the Root Mean Square Error of Approximation. 
  - TLI: Tucker-Lewis Index. Evaluates the fit compared to a more resticted, nested baseline model. Hopwood and Donnallan (2010) suggested that a value higher than .9 indicates a good fit. However, there is no consensus about this cutoff. 
  - BIC: Bayesian Information Criterion. This measure is useful for comparing the performances of different models on the same data, where a lower value indicates a better fitting model. 
  

> - Parallel Analysis: The table displays as many numbers of factors as variables selected for analysis, eigenvalues corresponding to the real-data factor, and the eigenvalue corresponding to the parallel mean resampled value. It will display an asterisk along the names of the factors advised to be retained (whose real-data eigenvalue is greater than the resampled-data mean value). Note that, even when selecting a PC-based parallel analysis, the table will refer to "factors" as the ones advised to be retained instead of "components"; this is due to common usage of the PC-based parallel analysis method for assessing the number of factors within EFA (e.g., Golino et al., 2020).

Reviewer: ```suggestion
- Parallel Analysis: The table displays as many factors as variables selected for analysis, eigenvalues corresponding to the real-data factor, and the eigenvalue corresponding to the parallel mean resampled value. It will display an asterisk along the names of the factors advised to be retained (whose real-data eigenvalue is greater than the resampled-data mean value). Note that, even when selecting a PC-based parallel analysis, the table will refer to "factors" as the ones advised to be retained instead of "components"; this is due to common usage of the PC-based parallel analysis method for assessing the number of factors within EFA (e.g., Golino et al., 2020).
```
